Add iconLibrary options to content-actions

c3dr0x opened the pull request:
Fix: #1467

### Description

* **primary**: now accepts an `iconLibrary` option to change icon library.
* **secondary**: can now be an object with keys similar to **primary**: `icon`, `iconLibrary`, `action`. 

```js
{
    primary: [
        { icon, iconLibrary, label, className, action },
        // or
        { icon, iconLibrary, action: [
            { label, action, style },
            [...]
        ]},
        [...]
    ],
    secondary: [
        { label, action, style },
        [...]
    ],
    // or
    secondary: { icon, iconLibrary, action: [
        { label, action, style },
        [...]
    ]}
}
```
note: `style` prop in action array is only available in legacy component.

Hartorn commented on src/application/content-actions/index.js:
```diff
@@ -31,29 +39,56 @@ const ContentActions = {
         return (
             <div className={this._getStyleClassName()} data-focus='content-actions'>
-                {actions.primary.map((primary) => {
+                {primary.map((primary) => {
```
Dont re-use same name for iterator and list element

Hartorn commented on src/application/content-actions/index.js:
```diff
@@ -31,29 +39,56 @@ const ContentActions = {
                     }
                 })}
-                <Dropdown iconProps={{ name: 'more_vert' }} operationList={actions.secondary} shape='fab' />
+                {isArray(secondary) && (
```
Basically, the code for rendering secondary or primary actions should be the same.
Can you make a renderActions function, and use it for both ?

Hartorn commented:
@c3dr0x Should work now. I let you test it before merging

c3dr0x commented:
- I refactored it in function style (no state anymore) 
- I made some improvements onto icon-dropdown for merging default props: `more_vert` icon wasn't displayed
- I fixed CSS when two icon-dropdowns where shown next to one another. They weren't displayed in the right order -> secondary was displayed in 2nd **before** the icon-dropdown from primary

My test :

```jsx
// Libs
import React from 'react';
// Focus
import cartridgeBehaviour from 'focus-components/page/mixin/cartridge-behaviour';

export default React.createClass({
    displayName: 'HomeView',

    mixins: [cartridgeBehaviour],

    getInitialState() {
        return {
            visible: false
        };
    },

    cartridgeConfiguration() {
        const cartridge = {
            canDeploy: true,
            actions: {
                primary: [],
                secondary: []
            }
        };

        // Primary action
        cartridge.actions.primary.push({
            label: 'action',
            icon: 'home',
            className: 'totoro',
            action: () => { alert('action') }
        });

        // Primary menu
        cartridge.actions.primary.push({
            label: 'menu',
            icon: 'cached',
            action: [{
                label: 'sub1',
                action: () => { alert('sub1') }
            }]
        });

        // Secondary classic
        // cartridge.actions.secondary = [{
        //     label: 'sub2',
        //     action: () => { alert('sub2') }
        // }];

        // Secondary new 
        cartridge.actions.secondary = {
            icon: 'close',
            action: [{
                label: 'sub3',
                action: () => { alert('sub3') }
            }]
        };

        return cartridge;
    },

    /** @inheritDoc */
    render() {
        return (
            <div data-app='homepage'>
                {'Home'}
            </div>
        );
    }
});

```
